mathesar_ui/src/systems/table-view/table-inspector/column/ColumnOptions.svelte

```diff
@@ -58,26 +58,30 @@
 
   async function toggleAllowDuplicates() {
     isRequestingToggleAllowDuplicates = true;
-    try {
-      const newAllowsDuplicates = !allowsDuplicates;
-      await constraintsDataStore.setUniquenessOfColumn(
-        column.column,
-        !newAllowsDuplicates,
-      );
-      const message = `Column "${column.column.name}" will ${
-        newAllowsDuplicates ? '' : 'no longer '
-      }allow duplicates.`;
-      toast.success({ message });
-      dispatch('close');
-    } catch (error) {
-      const message = `Unable to update "Allow Duplicates" of column "${
-        column.column.name
-        // @ts-ignore: https://github.com/centerofci/mathesar/issues/1055
-      }". ${error.message as string}.`;
-      toast.error({ message });
-    } finally {
-      isRequestingToggleAllowDuplicates = false;
-    }
+    const newAllowsDuplicates = !allowsDuplicates;
+    return constraintsDataStore
+      .setUniquenessOfColumn(column.column, !newAllowsDuplicates)
+      .then(async () => {
+        await columnsDataStore.patch(column.id, {
+          default: null,
+        });
+        const message = `Column "${column.column.name}" will ${
+          newAllowsDuplicates ? '' : 'no longer '
+        }allow duplicates.`;
+        toast.success({ message });
+        dispatch('close');
+        return null;
+      })
+      .catch((error) => {
+        const message = `Unable to update "Allow Duplicates" of column "${
+          column.column.name
+          // @ts-ignore: https://github.com/centerofci/mathesar/issues/1055
+        }". ${error.message as string}.`;
+        toast.error({ message });
+      })
+      .finally(() => {
+        isRequestingToggleAllowDuplicates = false;
+      });
```

Comment:
As per the FE standards here: https://github.com/centerofci/mathesar/blob/develop/mathesar_ui/STANDARDS.md#await-vs-then, async/await is preferred here. Was there any specific reason to change?